File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.py
class Solution:
    def findDuplicates(self, nums: List[int]) -> List[int]:
        nums.sort()
        ans = []
        temp = 1
        for i in range(1,len(nums)):
            if nums[i]==nums[i-1]:
                temp+=1
            else:
                if temp == 2: ans.append(nums[i-1])
                temp=1
        if temp ==2: ans.append(nums[-1])
        return ans
            
        

File: 0442-find-all-duplicates-in-an-array/0442-find-all-duplicates-in-an-array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        unordered_set<int> m;
        vector<int> ans;
        for(int val:nums){
            if(m.contains(val)){
                ans.push_back(val);
            }else{
             m.insert(val);   
            }
        }
        return ans;
    }
};